src/app/shared/services/data/app-data.service.ts

```diff
@@ -81,7 +81,7 @@ export class AppDataService extends SyncServiceBase {
     flow_type: FlowTypes.FlowType,
     flow_name: string,
     /** Keep log of previous override flow names to avoid infinite loops, e.g. flow_a -> flow_b -> flow_a  */
-    overrideHistory: string[] = []
+    is_override_target: boolean = false
```

Comment:
nit(blocking)
I'm a bit unsure about this, as I think making it binary would prevent an otherwise fine case of nested overrides (like in the comment). Would it be possible to add tests to the service spec file and confirm if the behaviour is as-expected or not?

Reply:
I think I wasn't clear on what behaviour tracking the `overrideHistory` was supposed to handle, and had it confused with the core use case of the PR's associated issue. But I think now I see the difference: `overrideHistory` is intended to handle the case where `flow_a` is overridden by `flow_b` which in turn is overridden by `flow_a`.

I've updated the PR with [changes to the app data service](https://github.com/IDEMSInternational/parenting-app-ui/pull/2107/files#diff-d5415ab90f2bbaee6aba223424f7e14faef364f1f635f0db46909bf5bdba7e70) which I think handle both cases, and [added tests](https://github.com/IDEMSInternational/parenting-app-ui/pull/2107/files#diff-3e326d5bce4e0471362869d3f6fed1af63ce3de890634b1ce0e9c91c618118f2) to capture the behaviour.